pkg/kubelet/kubelet.go

```diff
@@ -1311,7 +1311,14 @@ func (kl *Kubelet) syncPod(o syncPodOptions) error {
 	kl.statusManager.SetPodStatus(pod, apiPodStatus)
 
 	// Kill pod if it should not be running
-	if errOuter := canRunPod(pod); errOuter != nil || pod.DeletionTimestamp != nil || apiPodStatus.Phase == api.PodFailed {
+	errOuter := canRunPod(pod)
+	if errOuter == nil {
+		rs := kl.runtimeState.networkErrors()
+		if len(rs) != 0 && !podUsesHostNetwork(pod) {
+			errOuter = fmt.Errorf("Network is not ready: %v", rs)
```

Comment:
Fixed the error nit - thanks.

Can you explain when we do & don't want to kill the pod?  Is it the case that we kill if the failure is permanent, but not if the failure is transient?